src/probabilities_estimators/permutation_ordinal/spatial_permutation.jl

```diff
@@ -9,24 +9,52 @@ The input data `x` are high-dimensional arrays, for example 2D arrays [^Ribeiro2
 [^Schlemmer2018]. This approach is also known as _spatiotemporal permutation entropy_.
 `x` is given because we need to know its size for optimization and bound checking.
 
-A _stencil_ defines what local area around each pixel to
-consider, and compute the ordinal pattern within the stencil. Stencils are given as
-vectors of `CartesianIndex` which encode the _offsets_ of the pixes to include in the
-stencil, with respect to the current pixel. For example
-
-```julia
-data = [rand(50, 50) for _ in 1:50]
-x = data[1] # first "time slice" of a spatial system evolution
-stencil = CartesianIndex.([(0,1), (1,1), (1,0)])
-est = SpatialSymbolicPermutation(stencil, x)
-```
-
-Here the stencil creates a 2x2 square extending to the bottom and right of the pixel
-(directions here correspond to the way Julia prints matrices by default).
-Notice that no offset (meaning the pixel itself) is always included automatically.
-The length of the stencil decides the order of the permutation entropy, and the ordering
-within the stencil dictates the order that pixels are compared with.
-The pixel without any offset is always first in the order.
+A _stencil_ defines what local area (which points) around each pixel to
+consider, and compute ordinal patterns from.
+Stencils are passed in one of the following three ways:
+
+1. As vectors of `CartesianIndex` which encode the pixels to include in the
+    stencil, with respect to the current pixel, or integer arrays of the same dimensionality
+    as the data. For example
+
+    ```julia
+    data = [rand(50, 50) for _ in 1:50]
+    x = data[1] # first "time slice" of a spatial system evolution
+    stencil = CartesianIndex.([(0,0), (0,1), (1,1), (1,0)])
+    est = SpatialSymbolicPermutation(stencil, x)
+    ```
+    Don't forget to include the zero offset index if you want to include the point itself, 
+    which is almost always the case.
+    Here the stencil creates a 2x2 square extending to the bottom and right of the pixel
+    (directions here correspond to the way Julia prints matrices by default).
+    The number of included points in a stencil (`m`) determines the length of the vectors
```

Comment:
I think maybe this sentence *"The number of included points in a stencil (`m`) determines the length of the vectors to be symbolized, i.e. there are `m!` possible ordinal patterns around each pixel."* belongs *before* the enumerated list, because this is a statement that is true of all three stencil types. Then we should say how `m` is computed for each stencil type under the respective enumerated point.

For method 1, `m = length(stencil)`. For method 2 `m = count(!iszero, stencil)`. Is it possible to say what `m` would be a priori for method 3 too, @ikottlarz ?

Reply:
Yes, for the 3rd method it's just `m = prod(extent)!`. I'll add it to the docstring!

Reply:
@ikottlarz Does `prod(extent)!` also take into consideration the `lags`?